### ISSUE TITLE
add code QR option

### ISSUE THREAD

**Issue by alancivale** [Aymkdn/html-to-pdfmake#102]
I use the code tag and I clarify what type of code it is. I give you an example of use and the code to add so that I can take it in the options.

example:
```html
<code typecode="QR" style="foreground:black;background:yellow;fit:300px">texto in code</code>
```

```js
// add after of switch(nodeName) { 
case "CODE": {
                ret = applyStyle({ret:ret, parents:parents.concat([element])});
                ret.qr = ret.text[0].text;
                switch(element.getAttribute("typecode")){
                  case 'QR':
                    delete ret.text;
                    ret.nodeName='QR';
                    if(!ret.style || !Array.isArray(ret.style)){
                      ret.style = [];
                    }
                    ret.style.push('html-qr');
                    console.log(ret);
                    break;
                }
                break;
              }
```

I'm going to build more types of codes, the qr is included in the pdfmake, the others, like code128, have to import the source, and do the conversion. I'll give it to you later if I put it together

saludos!

**Comment by alancivale** [comment]
`
<code typecode="QR" style="foreground:black;background:yellow;fit:300px">texto in code</code>`

**Comment by Aymkdn** [comment]
This package should handle regular HTML code. However I may provide a way to manage this edge case.

I'll think about it tomorrow (I'm not home today) and I'll get back to you by the end of Sunday.

**Comment by alancivale** [comment]
Yes, calm, I know that it is quite a bit more, but since I am putting together impressions of things that may have barcode, I added the functionality. ean13 happened to you here. These I pass them loose, because it changes a lot when making it asynchronous and it would no longer be compatible for the one who used it previously, since you have to change how it is instantiated and called. but so I can download images and build a cache. Greetings and have a good weekend

```
<code typecode="EAN13" style="font-size: 300px;">123465</code>
<code typecode="EAN13" style="foreground:black;background:yellow;">0123456789012</code>
<code typecode="EAN13">012345678</code>

```
```
  case 'EAN13':
                    ret.nodeName='TEXT';
                    ret.text = ret.text[0].text;
                    if(!ret.style || !Array.isArray(ret.style)) {
                      ret.style = [];
                    }
                    if(!Number.isNaN(ret.text)) {
                      ret.text = toean13(ret.text);
                      ret.font = 'Ean13';
                      if(!ret.fontSize) {
                        ret.fontSize= 70;  
                      }
                    }
                    ret.style.push('html-ean13');
                    break;
```
```

    var toean13 =  function toean13(txt) {
      var  gug = ["LLLLLL","LLGLGG","LLGGLG","LLGGGL","LGLLGG","LGGLLG","LGGGLL","LGLGLG","LGLGGL","LGGLGL"];
      if(!txt){
        return txt;
      }
      var first = txt[0]
      var enc = "_"+first+"*"
      var max = ((txt.length<7)? txt.length:7)
      for(var i = 1;i< max;i++){
        enc+=gug[first][i-1]+txt[i]
      }

      enc+="**"

      // Right Block
      var max = ((txt.length<12)? txt.length:12)
      for(var i = 7;i<  max;i++){
        enc+="R"+txt[i]
      }
      var pr = 0

      //Checksum
      for(var i= Math.min(txt.length,12);i>=1;i--){
        pr+=parseInt(txt[i-1])*(i%2==1?1:3)
      }
      pr=(10-(pr%10))%10
      enc+="R"+pr
      enc+="*"

      return enc
    }

```

**Comment by Aymkdn** [comment]
I've just published v2.2.0 that can [handle custom tags](https://github.com/Aymkdn/html-to-pdfmake#customtag).

For your QR Code case:
```js
var html = htmlToPdfMake(`<code typecode="QR" style="foreground:black;background:yellow;fit:300px">texto in code</code>`, {,
  customTag:function(params) {
    var ret = params.ret;
    var element = params.element;
    var parents = params.parents;
    switch(ret.nodeName) {
      case "CODE": {
        ret = this.applyStyle({ret:ret, parents:parents.concat([element])});
        ret.qr = ret.text[0].text;
        switch(element.getAttribute("typecode")){
          case 'QR':
            delete ret.text;
            ret.nodeName='QR';
            if(!ret.style || !Array.isArray(ret.style)){
              ret.style = [];
            }
            ret.style.push('html-qr');
            break;
        }
        break;
      }
    }
    return ret;
  }
});
```
